### PR TITLE
Adding sleep functionality to MCU, accelerometer, and screen

### PR DESCRIPTION
**Pull request by mmoskal**

### CODE REVIEW

**Comment by mmoskal** [comment]
yeah, so setSleep() in CodalComponent (because you also need wake up)

**Comment by jamesadevine** [comment]
Sounds good :smile:

File: source/core/CodalComponent.cpp [hunk 1]
```diff
@@ -33,6 +33,10 @@ CodalComponent* CodalComponent::components[DEVICE_COMPONENT_COUNT];
 
 uint8_t CodalComponent::configuration = 0;
 
+#if DEVICE_COMPONENT_COUNT > 255
+#error "DEVICE_COMPONENT_COUNT has to fit in uint8_t"
```
Comment (mmoskal) [inline source/core/CodalComponent.cpp]: this is for existing code that uses uint8 for some reason

**Comment by mmoskal** [comment]
@jamesadevine this should be good to go

File: source/core/CodalComponent.cpp (hunk 1 above)
Comment (jamesadevine) [reply source/core/CodalComponent.cpp]: Weird!

Comment (mmoskal) [reply source/core/CodalComponent.cpp]: maybe we should just put the components on a linked list and pre-allocate an array for them. Would make iterating backwards harder, but I guess not too bad. Well, in future, when we have time...

**Pull request merged**